diff --git a/openmp-offload/io.c b/openmp-offload/io.c
index 5c981ee..c2fd0b7 100644
--- a/openmp-offload/io.c
+++ b/openmp-offload/io.c
@@ -275,8 +275,16 @@ Inputs read_CLI( int argc, char * argv[] )
 	{
 		char * arg = argv[i];
 
+		// This will be ignored
+		if( strcmp(arg, "-t") == 0 )
+		{
+			if( ++i < argc )
+				input.nthreads = atoi(argv[i]);
+			else
+				print_CLI_error();
+		}
 		// n_gridpoints (-g)
-		if( strcmp(arg, "-g") == 0 )
+        else if( strcmp(arg, "-g") == 0 )
 		{	
 			if( ++i < argc )
 			{
