diff -Naur rodinia_3.1/openmp/myocyte/Makefile rodinia_3.1-patched/openmp/myocyte/Makefile
--- rodinia_3.1/openmp/myocyte/Makefile	2015-12-11 07:46:20.000000000 -0800
+++ rodinia_3.1-patched/openmp/myocyte/Makefile	2020-02-04 12:55:46.000000000 -0800
@@ -7,10 +7,14 @@
           # .
 	# command n
 
+CC=gcc
+CFLAGS=-O0 -g -fopenmp
+LDFLAGS=-fopenmp
+
 # link objects(binaries) together
 myocyte.out:	main.o
-	gcc	main.o \
-			-lm -fopenmp \
+	$(CC)	main.o \
+			-lm $(CFLAGS) $(LDFLAGS) \
 	-o myocyte.out
 
 # compile main function file into object (binary)
@@ -24,9 +28,9 @@
 				solver.c \
 				file.c \
 				timer.c
-	gcc	main.c \
-			-c -O3 -fopenmp
+	$(CC)	main.c \
+			-c $(CFLAGS)
 
 # delete all object files
 clean:
-	rm *.o myocyte.out output.txt
+	rm -rf *.o myocyte.out output.txt
diff -Naur rodinia_3.1/openmp/myocyte/main.c rodinia_3.1-patched/openmp/myocyte/main.c
--- rodinia_3.1/openmp/myocyte/main.c	2015-12-11 07:46:20.000000000 -0800
+++ rodinia_3.1-patched/openmp/myocyte/main.c	2020-02-04 12:55:46.000000000 -0800
@@ -360,7 +360,7 @@
 	//================================================================================80
 
 	printf("Time spent in different stages of the application:\n");
-	printf("%.12f s, %.12f % : SETUP VARIABLES, READ COMMAND LINE ARGUMENTS\n", 	(float) (time1-time0) / 1000000, (float) (time1-time0) / (float) (time5-time0) * 100);
+	printf("%.6f s, %.6f % : SETUP VARIABLES, READ COMMAND LINE ARGUMENTS\n", 	(float) (time1-time0) / 1000000, (float) (time1-time0) / (float) (time5-time0) * 100);
 	printf("%.12f s, %.12f % : ALLOCATE MEMORY\n", 														(float) (time2-time1) / 1000000, (float) (time2-time1) / (float) (time5-time0) * 100);
 	printf("%.12f s, %.12f % : READ DATA FROM FILES\n", 												(float) (time3-time2) / 1000000, (float) (time3-time2) / (float) (time5-time0) * 100);
 	printf("%.12f s, %.12f % : RUN COMPUTATION\n", 														(float) (time4-time3) / 1000000, (float) (time4-time3) / (float) (time5-time0) * 100);
